sssom/util.py

```diff
@@ -483,49 +486,38 @@ def _extract_enum_text(df: pd.DataFrame) -> pd.DataFrame:
 
 
 def merge_msdf(
-    msdf1: MappingSetDataFrame,
-    msdf2: MappingSetDataFrame,
+    *msdfs: MappingSetDataFrame,
     reconcile: bool = True,
 ) -> MappingSetDataFrame:
-    """Merge msdf2 into msdf1.
+    """Merge multiple MappingSetDataFrames into one.
 
-    :param msdf1: The primary MappingSetDataFrame
-    :param msdf2: The secondary MappingSetDataFrame
+    :param msdfs: A Tuple of MappingSetDataFrames to be merged
     :param reconcile: If reconcile=True, then dedupe(remove redundant lower confidence mappings)
         and reconcile (if msdf contains a higher confidence _negative_ mapping,
         then remove lower confidence positive one. If confidence is the same,
         prefer HumanCurated. If both HumanCurated, prefer negative mapping).
         Defaults to True.
     :returns: Merged MappingSetDataFrame.
     """
-    # Inject metadata of msdf into df
-    msdf1 = inject_metadata_into_df(msdf=msdf1)
-    msdf2 = inject_metadata_into_df(msdf=msdf2)
-
-    msdf1.df = _extract_enum_text(msdf1.df)
-    msdf2.df = _extract_enum_text(msdf2.df)
-
     merged_msdf = MappingSetDataFrame()
-    # If msdf2 has a DataFrame
-    if msdf1.df is not None and msdf2.df is not None:
-        # 'outer' join in pandas == FULL JOIN in SQL
-        merged_msdf.df = msdf1.df.merge(msdf2.df, how="outer")
-    else:
-        merged_msdf.df = msdf1.df
 
-    # merge the non DataFrame elements
-    merged_msdf.prefix_map = dict_merge(
-        source=msdf2.prefix_map,
-        target=msdf1.prefix_map,
-        dict_name="prefix_map",
+    # Inject metadata of msdf into df
+    msdf_with_meta = [inject_metadata_into_df(msdf) for msdf in msdfs]
```

Comment:
You iterate through all msdfs to be merged, and call a function "inject_metadata" that only has a single parameter.. Were is the metadata to be injected coming from?

Reply:
The metadata is in `msdf.metadata`. The code looks like this:

```
    if msdf.metadata is not None and msdf.df is not None:
        for k, v in msdf.metadata.items():
            if k not in msdf.df.columns:
                msdf.df[k] = v
    return msdf
```

Reply:
Ahhh now I remember this. Ok! Thanks.. This is why there is no metadata in the merged artefact.. Now it all makes sense. The metadata is moved to the columns..